src/swap/saga.ts

```diff
@@ -247,8 +250,182 @@ export function* swapSubmitSaga(action: PayloadAction<SwapInfo>) {
+  const toToken =
+    tokens[getTokenId(networkConfig.defaultNetworkId, action.payload.userInput.toToken)]
+
+  if (!fromToken || !toToken) {
+    Logger.error(
+      TAG,
+      `Could not find to or from token for swap from ${sellTokenAddress} to ${buyTokenAddress}`
+    )
+    yield* put(swapError())
+    return
+  }
+
+  const fromTokenBalance = fromToken.balance.shiftedBy(fromToken.decimals).toString()
+  const estimatedSellTokenUsdValue = calculateEstimatedUsdValue({
+    tokenInfo: fromToken,
+    tokenAmount: sellAmount,
+  })
+  const estimatedBuyTokenUsdValue = calculateEstimatedUsdValue({
+    tokenInfo: toToken,
+    tokenAmount: buyAmount,
+  })
+
+  const swapApproveContext = newTransactionContext(TAG, 'Swap/Approve')
+  const swapExecuteContext = newTransactionContext(TAG, 'Swap/Execute')
+
+  const defaultSwapExecuteProps = {
+    toToken: buyTokenAddress,
+    fromToken: sellTokenAddress,
+    amount,
+    amountType,
+    price,
+    allowanceTarget,
+    estimatedPriceImpact,
+    provider: action.payload.quote.rawSwapResponse.details.swapProvider,
+    fromTokenBalance,
+    swapExecuteTxId: swapExecuteContext.id,
+    swapApproveTxId: swapApproveContext.id,
+    estimatedSellTokenUsdValue,
+    estimatedBuyTokenUsdValue,
+  }
+
+  let quoteToTransactionElapsedTimeInMs: number | undefined
+
+  const getTimeMetrics = (): SwapTimeMetrics => ({
+    quoteToUserConfirmsSwapElapsedTimeInMs: swapSubmittedAt - quoteReceivedAt,
+    quoteToTransactionElapsedTimeInMs,
+  })
+
+  try {
+    // Navigate to swap pending screen
+    navigate(Screens.SwapExecuteScreen)
+
+    const wallet = yield* call(getViemWallet, networkConfig.viemChain.celo)
+    if (!wallet.account) {
+      // this should never happen
+      throw new Error('no account found in the wallet')
+    }
+
+    const preparedTransactions = action.payload.quote.preparedTransactions
+    if (preparedTransactions?.type !== 'possible') {
+      // Should never happen
+      throw new Error('No prepared transactions possible')
+    }
+
+    // @ts-ignore typed-redux-saga erases the parameterized types causing error, we can address this separately
+    let nonce: number = yield* call(getTransactionCount, wallet, {
+      address: wallet.account.address,
+      blockTag: 'pending',
+    })
+
+    // Unlock account before executing tx
+    yield* call(unlockAccount, wallet.account.address)
+
+    // Execute transaction(s)
+    yield* put(swapExecute())
+    Logger.debug(TAG, `Starting to swap execute for address: ${wallet.account.address}`)
+
+    const beforeSwapExecutionTimestamp = Date.now()
+    quoteToTransactionElapsedTimeInMs = beforeSwapExecutionTimestamp - quoteReceivedAt
+
+    const txHashes: Hash[] = []
+    for (const preparedTransaction of preparedTransactions.transactions) {
+      const signedTx = yield* call([wallet, 'signTransaction'], {
+        ...preparedTransaction,
+        nonce: nonce++,
+      } as any)
```

Comment:
Casting here is also because of `typed-redux-saga` erasing parameterized types and causing errors.
I have local branch where I tried to solve this, I'll try to finish it.